frame/democracy/src/lib.rs

```diff
@@ -935,8 +1017,17 @@ decl_module! {
 		/// Emits `Delegated`.
 		///
 		/// # <weight>
+		/// - Complexity: `O(R)` where R is the number of referendums the voter delegating to has
+		///   voted on. Weight is charged as if maximum votes.
+		/// - Db reads: 2*`VotingOf`, `balances locks`
+		/// - Db writes: 2*`VotingOf`, `balances locks`
+		/// - Db reads per votes: `ReferendumInfoOf`
+		/// - Db writes per votes: `ReferendumInfoOf`
+		// NOTE: weight must cover an incorrect voting of origin with 100 votes.
 		/// # </weight>
-		#[weight = 500_000_000]
+		#[weight = 82_000_000 + 8_800_000 * Weight::from(T::MaxVotes::get())
+			+ T::DbWeight::get().reads_writes(3 + Weight::from(T::MaxVotes::get()), 3 + Weight::from(T::MaxVotes::get()))]
+		// TODO TODO: this might be too expensive, at least we should refund maybe.
```

Comment:
agreed

Reply:
I implemented for the most costy ones, (all delegate ones)